src/main/java/io/openapitools/swagger/GenerateMojo.java

```diff
@@ -127,6 +136,49 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         });
     }
 
+    private Application resolveApplication(JaxRSScanner reflectiveScanner) {
+        if (applicationClass == null || applicationClass.isEmpty()) {
+            Set<Class<? extends Application>> appClasses = reflectiveScanner.applicationClasses();
+            if (appClasses.isEmpty()) {
+                return null;
+            }
+            if (appClasses.size() > 1) {
+                getLog().warn("More than one javax.ws.rs.core.Application classes found on the classpath, skipping");
+                return null;
+            }
+            return instantiateApplication(appClasses.iterator().next());
+        }
+
+        Class<?> clazz = resolveClass(applicationClass);
+
+        if (!Application.class.isAssignableFrom(clazz)) {
+            getLog().warn("Provided application class does not implement javax.ws.rs.core.Application, skipping");
+            return null;
+        }
+
+        @SuppressWarnings("unchecked")
+        Class<? extends Application> appClazz = (Class<? extends Application>)clazz;
+        return instantiateApplication(appClazz);
+    }
+    
+    private Class<?> resolveClass(String className) {
+        try {
+            return Class.forName(applicationClass, true, Thread.currentThread().getContextClassLoader());
+        } catch (ClassNotFoundException e) {
+            getLog().warn("Cannot load class: " + applicationClass);
+            return null;
+        }
+    }
+
+    private Application instantiateApplication(Class<? extends Application> clazz) {
+        try {
+            return clazz.getDeclaredConstructor().newInstance();
```

Comment:
I have not checked but does JAX-RS mandate that the `Application` instance must have a no-arg constructor? I am unsure as to whether you could do constructor based injections if using JAX-RS in a CDI context.

Reply:
I have checked the JAX-RS specification ([here](https://download.oracle.com/javaee-archive/jax-rs-spec.java.net/jsr339-experts/att-3593/spec.pdf)) which mentions in section 10.2.7 that:

"Support for constructor injection of JAX-RS resources is OPTIONAL. Portable applications MUST
instead use fields or bean properties in conjunction with a `@PostConstruct` annotated method. Implementations SHOULD warn users about use of non-portable constructor injection."

I have tried this on Wildfly (made a CDI injectable constructor in Application class, which was itself an `@ApplicationScoped` CDI bean), and Resteasy did not like it (deployment containing such Application class could not be deployed). 

So I would say that we can assume that the Application class must have accessible no-arg constructor.

Reply:
great - then we should be home-safe on that one :+1: